pkg/kube/client.go

```diff
@@ -252,22 +252,25 @@ func (c *Client) Update(original, target ResourceList, force bool) (*Result, err
 // ResourceList that is part of the result.
 func (c *Client) Delete(resources ResourceList) (*Result, []error) {
 	var errs []error
+	var err error
 	res := &Result{}
 	mtx := sync.Mutex{}
-	err := perform(resources, func(info *resource.Info) error {
-		c.Log("Starting delete for %q %s", info.Name, info.Mapping.GroupVersionKind.Kind)
-		if err := c.skipIfNotFound(deleteResource(info)); err != nil {
-			mtx.Lock()
-			defer mtx.Unlock()
-			// Collect the error and continue on
-			errs = append(errs, err)
-		} else {
-			mtx.Lock()
-			defer mtx.Unlock()
-			res.Deleted = append(res.Deleted, info)
-		}
-		return nil
-	})
+	if len(resources) > 0 {
+		err = perform(resources, func(info *resource.Info) error {
+			c.Log("Starting delete for %q %s", info.Name, info.Mapping.GroupVersionKind.Kind)
+			if err := c.skipIfNotFound(deleteResource(info)); err != nil {
+				mtx.Lock()
+				defer mtx.Unlock()
```

Comment:
appears the mutex lock can happen above the `if err` statement so its not repeated

Reply:
I have moved the check to uninstall instead of the client.